### ISSUE TITLE
Is celery 4.x compatible with python 3.6?

### ISSUE THREAD

**Issue by thoas** [celery/celery#4000]
Hi,

I'm trying to use celery 4.0.2 with python 3.6.1 and the worker is exiting immediately without any stacktrace which is kind of frustrating for me because I'm not able to provide it in this issue.

**OUTPUT**

```console
[2017-04-26 15:59:28,066: INFO/MainProcess] Connected to amqp://guest:**@127.0.0.1:5672/guest
[2017-04-26 15:59:28,077: INFO/MainProcess] mingle: searching for neighbors
[2017-04-26 15:59:29,100: INFO/MainProcess] mingle: all alone
[2017-04-26 15:59:30,125: DEBUG/MainProcess] removing tasks from inqueue until task handler finished
make: *** [run-worker] Error 1
```

**SETTINGS**

```python
CELERY_RESULT_BACKEND = 'amqp'
CELERY_TASK_RESULT_EXPIRES = 300 # 5 minutes
CELERY_EVENT_QUEUE_TTL = 300 # 5 minutes
CELERY_DEFAULT_QUEUE = 'default'
CELERY_SEND_TASK_ERROR_EMAILS = False
CELERY_DEFAULT_RATE_LIMIT = 300
CELERY_ACCEPT_CONTENT = ['json', 'application/json']
CELERY_TASK_SERIALIZER = 'json'
CELERY_RESULT_SERIALIZER = 'json'
```
**VERSIONS**

```console
amqp==2.1.4
billiard==3.5.0.2
celery==4.0.2
kombu==4.0.2
```

I cannot reproduce this issue in python 3.5.2 which is working well.

**Comment by georgepsarakis** [comment]
Travis CI is running the unit and integration tests with Python 3.6. This Python version was introduced in the current master, so perhaps there are some issues with 4.0.2.

**Comment by thoas** [comment]
Thank you for the answer, it's quite funny since we are now using the celery 3.x on python 3.6 and it works like a charm :)

**Comment by andreyrusanov** [comment]
I use Celery 4.0.2 and Python 3.6 - works without issues

**Comment by soon** [comment]
If anyone else is facing similar issues with celery, try to remove queue. In my case it was

    rabbitmqadmin delete queue name=default 

I do not know the exact issue, probably I mixed celery with dramatiq locally (i.e. they both refer to the same queue) and maybe dramatiq creates queue in a different way - do not know. But I tried to downgrade celery to 4.0, 4.1 and 4.2 including their minor versions - none of them works. Removing the queue fixed the issue.
